pkg/controllers/management/rbac/legacy_ownerrefs_cleaner.go

```diff
@@ -0,0 +1,214 @@
+package rbac
+
+import (
+	"context"
+
+	typesv1 "github.com/rancher/rancher/pkg/generated/norman/core/v1"
+	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	v1 "github.com/rancher/rancher/pkg/generated/norman/rbac.authorization.k8s.io/v1"
+	namespaces "github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/retry"
+)
+
+const (
+	roleOwnerRefCorrected        = "auth.management.cattle.io/role-owner-ref-corrected"
+	roleBindingOwnerRefCorrected = "auth.management.cattle.io/role-binding-owner-ref-corrected"
+)
+
+type ownerRefCleaner struct {
+	mgmt         *config.ManagementContext
+	roles        v1.RoleInterface
+	roleBindings v1.RoleBindingInterface
+}
+
+func Register(ctx context.Context, m *config.ManagementContext) {
+	o := ownerRefCleaner{
+		mgmt:         m,
+		roles:        m.RBAC.Roles(""),
+		roleBindings: m.RBAC.RoleBindings(""),
+	}
+	o.roles.AddHandler(ctx, "legacy-role-ownerref-cleaner", o.roleSync)
+	o.roleBindings.AddHandler(ctx, "legacy-rolebinding-ownerref-cleaner", o.roleBindingSync)
+}
+
+/* Following objects had Roles and RoleBindings with incorrect OwnerRefs prior to 2.4.6:
+Cloud Credential (Secret representing a cloud credential)
+Node Template
+Cluster Template
+Cluster Template Revision
+Global DNS Entry
+Global DNS Provider
+Multi Cluster App
+Multi Cluster App Revision
```

Comment:
Move this to a package level comment above the package name to explain what this is doing